extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -48,19 +51,31 @@
   private static final Logger LOG = new Logger(OpenCensusProtobufInputRowParser.class);
 
   private static final String SEPARATOR = "-";
-  public static final String NAME = "name";
-  public static final String VALUE = "value";
-  public static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String DEFAULT_METRIC_DIMENSION = "name";
+  private static final String VALUE = "value";
+  private static final String TIMESTAMP_COLUMN = "timestamp";
+  private static final String DEFAULT_RESOURCE_PREFIX = "resource.";
   private final ParseSpec parseSpec;
   private final List<String> dimensions;
 
+  private final String metricDimension;
+  private final String metricLabelPrefix;
+  private final String resourceLabelPrefix;
+
   @JsonCreator
   public OpenCensusProtobufInputRowParser(
-      @JsonProperty("parseSpec") ParseSpec parseSpec
+      @JsonProperty("parseSpec") ParseSpec parseSpec,
+      @JsonProperty("metricDimension") String metricDimension,
+      @JsonProperty("metricLabelPrefix") String metricPrefix,
+      @JsonProperty("resourceLabelPrefix") String resourcePrefix
   )
   {
     this.parseSpec = parseSpec;
     this.dimensions = parseSpec.getDimensionsSpec().getDimensionNames();
+    this.metricDimension = !Strings.isNullOrEmpty(metricDimension) ? metricDimension : DEFAULT_METRIC_DIMENSION;
+    this.metricLabelPrefix = NullHandling.nullToEmptyIfNeeded(metricPrefix);
+    this.resourceLabelPrefix = resourcePrefix != null ? resourcePrefix : DEFAULT_RESOURCE_PREFIX;
```

Comment:
why are we handling `resourceLabelPrefix` and `metricLabelPrefix` differently? They should both be treated the same way. What we do here seems perfectly fine for metricsLabelPrefix as well.

Reply:
As per the discussion we had I think we agreed to have `resource.` as default prefix for `resource` labels where `metric` labels will default to `empty`.

https://github.com/confluentinc/druid/pull/14#discussion_r406329904

Reply:
right, we could do this by explicitly having `DEFAULT_METRIC_PREFIX = ""` to make it explicit what the defaults are, but if you want to leave it as is that's fine too.

Reply:
Leaving it as it is for now. Will change if needed for some other default prefix.

Reply:
that's fine, it was mostly a suggestion to make the pattern more obvious to someone reading the code.